packages/elements/src/photon-medication-dropdown-full-width/photon-medication-dropdown-full-width.tsx

```diff
@@ -0,0 +1,490 @@
+
+setBasePath('https://cdn.jsdelivr.net/npm/@shoelace-style/shoelace@2.4.0/dist/');
+
+//Styles
+import shoelaceDarkStyles from '@shoelace-style/shoelace/dist/themes/dark.css?inline';
+import shoelaceLightStyles from '@shoelace-style/shoelace/dist/themes/light.css?inline';
+import tailwind from '../tailwind.css?inline';
+import styles from './style.css?inline';
+
+//Types
+import { Treatment, PrescriptionTemplate, TreatmentOption } from '@photonhealth/sdk/dist/types';
+
+//Virtual List
+import { createVirtualizer, VirtualItem } from '@tanstack/solid-virtual';
+
+// Disable the default dropdown animation
+// TODO: Remove when the dropdown is swapped for a regular list
+setDefaultAnimation('dropdown.show', null);
+
+interface DataItem<T> {
+  data: T;
+  // TODO: setting this to scroll to the correct index
+  // after reopening the list, but not working ATM
+  allItemsIdx: number;
+}
+
+interface GroupTitle {
+  title: string;
+}
+
+// Typescript and solid are annoying
+// eslint-disable-next-line @typescript-eslint/ban-types
+type ThisisNotAFunction<T> = Exclude<T, Function>;
+
+export const PhotonMedicationDropdownFullWidth = <
+  T extends { id: string } | { medicationId: string }
+>(props: {
+  data: Array<Treatment | PrescriptionTemplate | TreatmentOption>;
+  label?: string;
+  required: boolean;
+  placeholder?: string;
+  forceLabelSize?: boolean;
+  invalid?: boolean;
+  onSearchChange?: (search: string) => void;
+  displayAccessor: (selected: T, groupDisplay: boolean) => string | JSXElement;
+  disabled?: boolean;
+  onOpen?: () => void;
+  onHide?: () => void;
+  isLoading: boolean;
+  hasMore: boolean;
+  noDataMsg?: string;
+  helpText?: string;
+  fetchMore?: () => void;
+  selectedData?: T | undefined;
+  groups?: Array<{
+    label: string;
+    filter: (arr: T) => boolean | undefined;
+  }>;
+  showOverflow?: boolean;
+  optional?: boolean;
+  clearable?: boolean;
+  actionRef?: any;
+  open: boolean;
+  searchText: string;
+  onClose: () => void;
+}) => {
+  //refs
+  let ref: any;
+  let dropdownRef: any;
+  let listRef: HTMLDivElement | undefined;
+  let inputRef: any;
+  let overlayRef: HTMLDivElement | undefined;
+
+  if (props.actionRef) {
+    props.actionRef['clear'] = () => {
+      setSelected(undefined);
+      setSelectedIndex(-1);
+      inputRef.value = '';
+      debounceSearch('');
+      dispatchDeselect();
+    };
+  }
+
+  //signals
+  const [selected, setSelected] = createSignal<T | undefined>(undefined);
+  const [selectedIndex, setSelectedIndex] = createSignal(-1);
+
+  const debounceSearch = debounce(async (s: string) => {
+    if (props.onSearchChange) {
+      await props.onSearchChange(s);
+    }
+  }, 250);
+
+  createEffect(() => {
+    if (props.open) {
+      dropdownRef?.show();
+      handleDropdownShow();
+    }
+  });
+
+  createEffect(() => {
+    if (props.data.length === 0) {
+      // if the source data changes then reset the selected value
```

Comment:
Yeah it's funky, it was also residual, I want to look into creating a useStore hook so that we don't have to pass so much stuff down. Leaving for this iteration.